src/course-outline/paste-button/PasteButton.jsx

```diff
@@ -0,0 +1,113 @@
+import { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+import {
+  Hyperlink, Icon, Button, Popover, OverlayTrigger,
+} from '@edx/paragon';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import {
+  FileCopy as PasteIcon,
+  Question as QuestionIcon,
+} from '@edx/paragon/icons';
+import { getInitialUserClipboard } from '../data/selectors';
+import messages from './messages';
+
+const PasteButton = ({
+  text,
+  blockType,
+  onClick,
+}) => {
+  const intl = useIntl();
+  const initialUserClipboard = useSelector(getInitialUserClipboard);
+  const {
+    content,
+    sourceContextTitle,
+    sourceEditUrl,
+  } = initialUserClipboard || {};
+  // Show button only if clipboard has content
+  const showPasteButton = (
+    content?.status === 'ready'
+    && content?.blockType === blockType
+  );
+
+  const [show, setShow] = useState(false);
+  const handleOnMouseEnter = () => {
+    setShow(true);
+  };
+  const handleOnMouseLeave = () => {
+    setShow(false);
+  };
+  const ref = useRef(null);
+
+  if (!showPasteButton) {
+    return null;
+  }
+
+  const renderBlockLink = (props) => (
+    <Popover
```

Comment:
This element should be a `<ModalPopup>` because `<Popover>` should not contain interactive elements due to accessibility.

Reply:
Used `Popover` just for the arrow icon but `ModalPopup` doesn't seem to work nicely with hover events. Removed `Popover` and added the arrow icon via css and made the Hyperlink popup using overlay component directly. The div is accessible similar to how it was in legacy.

Reply:
Updated z-index.